### PR TITLE
Add support for connecting to Azure China and Azure GovCloud workspaces

### PR DESCRIPTION
**Pull request by fjakobs**
Tested against both Azure China and Azure GovCloud workspaces with PAT and AZ CLI.

Currently sync doesn't work with Az CLI. Needs a patch in bricks

### CODE REVIEW

File: packages/databricks-vscode/src/configuration/AzureCliCheck.ts [hunk 1]
```diff
@@ -12,6 +12,12 @@ import {AzureCliAuthProvider} from "./auth/AuthProvider";
 const extensionVersion = require("../../package.json")
     .version as ProductVersion;
 
+type AzureCloud =
+    | "AzureCloud"
+    | "AzureChinaCloud"
+    | "AzureUSGovernment"
+    | "AzureGermanCloud";
```
Comment (pietern) [inline packages/databricks-vscode/src/configuration/AzureCliCheck.ts:19]: Last enum looks unused.

Comment (fjakobs) [reply packages/databricks-vscode/src/configuration/AzureCliCheck.ts:19]: Actually let me check what the German URLs are before merging

Comment (fjakobs) [reply packages/databricks-vscode/src/configuration/AzureCliCheck.ts]: We are not in Azure Germany. Took out this enum value.

File: packages/databricks-sdk-js/src/config/Config.ts [hunk 2]
```diff
@@ -258,7 +258,10 @@ export class Config {
      */
     public isAzure(): boolean {
         return (
-            (!!this.host && this.host.endsWith(".azuredatabricks.net")) ||
+            (!!this.host &&
+                !!this.host.match(
+                    /(\.databricks\.azure\.us|\.databricks\.azure\.cn|\.azuredatabricks\.net)$/
```
Comment (pietern) [inline packages/databricks-sdk-js/src/config/Config.ts:263]: `!!` to boolify the result?

The regexp could be collapsed with `(us|cn)`.

**Review by pietern — APPROVED** [review]
LGTM

File: packages/databricks-sdk-js/src/config/Config.ts (hunk 2 above)
Comment (fjakobs) [reply packages/databricks-sdk-js/src/config/Config.ts:263]: Yes `!!` is a JS idiom to convert a truthy of falsy value to a proper boolean

File: packages/databricks-vscode/src/configuration/AzureCliCheck.ts [hunk 3]
```diff
@@ -298,6 +300,13 @@ export class AzureCliCheck implements Disposable {
     // check if Azure CLI is logged in
     public async isAzureCliLoggedIn(): Promise<boolean> {
         try {
+            await ExecUtils.execFileWithShell(this.azBinPath, [
+                "cloud",
+                "set",
+                "--name",
+                this.getAzureCloud(this.authProvider.host),
+            ]);
```
[the remark this answers is not in the document]
Comment (fjakobs) [reply packages/databricks-vscode/src/configuration/AzureCliCheck.ts:308]: Unfortunately it is. Although it does mean that you can be logged in to China and regular Azure at the same time ans switch without signing out.

**Pull request merged**